### ISSUE TITLE
[BUG] Model undeploying giving empty response

### ISSUE THREAD

**Issue by gaurav7830** [opensearch-project/ml-commons#3285]
**What is the bug?**
When the model is in partially deployed state and undeployed is being called on that domain than the undeploy is giving emtpy response.

**How can one reproduce the bug?**
1. Create a domain.
2. Update the model state to partially deployed state.
3. Call model undeploy api.
4. Response will be empty.

**What is the expected behavior?**
Model should be in undeployed state after calling undeploy api.

**What is your host/environment?**
OS 2.17 version

**Do you have any screenshots?**
None

**Do you have any additional context?**
None

**Comment by brianf-aws** [comment]
Hey Gaurav, Thank you for creating this issue I will look into it!

**Comment by brianf-aws** [comment]
Hey @gaurav7830 Im trying to replicate locally. but I can not seem to have that empty response even when I purposely bring down a node during deployement it seems smart enough to give back correct results.

Im wondering if you have the `plugins.ml_commons.sync_up_job_interval_in_seconds: (number)` set on your cluster
https://opensearch.org/docs/latest/ml-commons-plugin/cluster-settings/#set-sync-job-intervals

Currently I tested with the sync up as "disabled". But please let me know what interval you are using so I can closely get as close.

**Comment by brianf-aws** [comment]
You can get the scenario organically if do the following.

Suppose you have a OS cluster with nodes a,b,c,d

1. Deploy to all nodes a,b,c,d
2. Check the model status and also the profile API `GET {{ _.domain }}/_plugins/_ml/profile/models/{{ _.model_id }}` Notice that the model is deployed on all nodes worker_nodes : {a,b,c,d}
3. Bring down one node (Lets use node **a**). While this occurs make sure the sync up job is running for example 3 seconds should change state
4. After some time youll see that the model changes state to `PARTIALLY_DEPLOYED`
--  --
> At this point Ive theorized that the cache of node is volatile (i.e. It gets lost when the node gets shut down)

5. (Stop the sync up job) Bring back **a**, if you check the profile api youll notice that even though the the node is back the profile API states that only 3 nodes service the node.
-- --
> If you undeploy at this state you'll get back a valid response stating that it could undeploy from **{b,c,d}** and that node **a** had a not_found response
https://github.com/opensearch-project/ml-commons/blob/7209a10ae2021631234fc4bbe8101049145a6fcb/plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java#L1872-L1884

6. Bring down other nodes **{b,c,d}**  When you bring them back now youll see the edge case of PARTIALLY_DEPLOYED that is hard to work with which is why it returns a {} response so the models aren't found in their cache so it skips (Since nodes **{a,b,c,d}** all report failures of not found in cache)
https://github.com/opensearch-project/ml-commons/blob/618678ffba35f77848dfd8e743785ad01f8efcfb/plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java#L735-L741
https://github.com/opensearch-project/ml-commons/blob/d878fbdea505a62c1b60834512165de9e3aeb68f/plugin/src/main/java/org/opensearch/ml/action/undeploy/TransportUndeployModelAction.java#L102-L110

---
### In Summary
The reality is that here i've illustrated two edge cases of the PARTIALLY_DEPLOYED state (There may be more edge cases). In one scenario if nodes have a recollection of the model state, then undeploy can give back a accurate response. But if all nodes have been shut down then there is no way to check it can undeploy since the nodes dont have the model in their cache.
